static_frame/core/frame.py

```diff
@@ -5968,6 +6021,55 @@ def to_latex(self,
                 )
         return repr(self.display(config))
 
+    def to_msgpack(self) -> 'bin':
+        '''
+        Return a msgpack.
+        '''
+        import msgpack
+        import msgpack_numpy
+        def encode(obj, chain=msgpack_numpy.encode):
+            clsname = obj.__class__.__name__
+            package = obj.__class__.__module__.split('.',1)[0]
+            print('package, clsname', package, clsname)
+            if package == 'static_frame':
+                if clsname in ['Index', 'IndexBase']:
+                    return {b'sf':clsname,
+                            b'name':obj.name,
+                            b'data':packb(obj.values)} #recursively call packb to nest msgpacks within
+                elif clsname in ['IndexHierarchy', 'IndexHierarchyGO']:
+                    if obj._recache:
+                        obj._update_array_cache()
+                    return {b'sf':clsname,
+                            b'name':obj.name,
+                            b'data':packb(obj._blocks._blocks), #recursively call packb to nest msgpacks within
+                            b'levels':packb(obj._levels)} #recursively call packb to nest msgpacks within
+                elif clsname in ['IndexLevel']:
+                    return {b'sf':clsname,
+                            b'index':packb(obj.index), #recursively call packb to nest msgpacks within
+                            b'_depth':obj._depth,
+                            b'_length':obj._length}
+                elif clsname in ['Frame']:
+                    return {b'sf':clsname,
+                            b'name':obj.name,
+                            b'data':packb(obj._blocks._blocks), #recursively call packb to nest msgpacks within
+                            b'index':packb(obj.index), #recursively call packb to nest msgpacks within
+                            b'columns':packb(obj.columns)} #recursively call packb to nest msgpacks within
+                elif clsname in ['IndexDate']:
+                    return {b'sf':clsname,
+                           }
+            if package == 'numpy':
+                if isinstance(obj, np.ndarray):
+                    if isinstance(obj[0], np.datetime64) or isinstance(obj[0], np.timedelta64):
+                        #TODO: Couldn't find an attribute for unit, just splitting it from the name for now
+                        unit = str(obj[0].dtype).split('[',1)[-1].split(']',1)[0]
+                        data = [[int(a.astype(int)), unit] for a in obj]
```

Comment:
Can we just convert the entire array at once, i.e., `obj.astype(int)`, and then store the unit in a different field of your dictionary? It will be a lot faster than doing the list comprehension.

Reply:
Doing that now using the key b'dtype'